ClientCore/ProfanityFilter.cs

```diff
@@ -1,90 +1,97 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Text.RegularExpressions;
+using System.Linq;
+using System.Text;
+using System.Diagnostics;
 
 namespace ClientCore
 {
     public class ProfanityFilter
     {
-        public IList<string> CensoredWords { get; private set; }
+        private const char CENSOR_CHAR = '*';
+        private readonly Regex _combinedRegex;
 
         /// <summary>
         /// Creates a new profanity filter with a default set of censored words.
         /// </summary>
-        public ProfanityFilter()
-        {
-            CensoredWords = new List<string>()
+        /// <param name="extraWords">Any extra words to be considered profane.</param>
+        /// 
+        public ProfanityFilter(IEnumerable<string> extraWords = null)
+        {        
+            var defaultWords = new[]
             {
-                "cunt*",
-                "*nigg*",
-                "paki*",
-                "shit",
-                "fuck*",
-                "admin*",
-                "allahu*",
-                "akbar",
-                "twat",
-                "cock",
-                "pussy",
-                "hitler*",
-                "anal"
+                "fuck", "shit", "cunt", "nigger", "nigga", "niggr", "cock",
+                "hitler", "pussy", "akbar", "allahu", "paki", "twat"
```

Comment:
With client being localized now, I think it's high time to add localization support here somehow